atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TracingCassandraClient.java

```diff
@@ -180,13 +186,19 @@ public CASResult cas(
     public CqlResult execute_cql3_query(CqlQuery cqlQuery, Compression compression, ConsistencyLevel consistency)
             throws InvalidRequestException, UnavailableException, TimedOutException, SchemaDisagreementException,
                     TException {
-        try (CloseableTrace trace =
-                startLocalTrace("cqlExecutor.execute_cql3_query(query {})", cqlQuery.getLazySafeLoggableObject())) {
+        try (CloseableTracer trace = startLocalTrace("cqlExecutor.execute_cql3_query", sink -> {
+            sink.accept("query", cqlQuery.getSafeLog());
+        })) {
             return client.execute_cql3_query(cqlQuery, compression, consistency);
         }
     }
 
-    private static CloseableTrace startLocalTrace(CharSequence operationFormat, Object... formatArguments) {
-        return CloseableTrace.startLocalTrace(SERVICE_NAME, operationFormat, formatArguments);
+    private static CloseableTracer startLocalTrace(
+            @CompileTimeConstant String operation, Consumer<TagConsumer> tagTranslator) {
+        return CloseableTracer.startSpan(getOperation(operation), FunctionalTagTranslator.INSTANCE, tagTranslator);
+    }
+
+    private static String getOperation(@CompileTimeConstant String operation) {
+        return SERVICE_NAME + "." + operation;
```

Comment:
If the inputs are compile-time-constants, should we inline the service name into those constants rather than re-concatenating each call? This runs even for unsampled traces.

Reply:
I am happy to do that, if the compiler is not smart enough. Lemme push a commit

Reply:
Done